lib/matplotlib/ticker.py

```diff
@@ -2658,9 +2658,9 @@ def __call__(self):
             majorstep_no_exponent = 10 ** (np.log10(majorstep) % 1)
 
             if np.isclose(majorstep_no_exponent, [1.0, 2.5, 5.0, 10.0]).any():
-                ndivs = 5
+                ndivs = rcParams['xtick.minor.default'] + 1
```

Comment:
I don't think this generalization is the way to go. This distinction is specifically designed to give reasonable intervals for certain interval steps, e.g.
for a majorstep of 10, you get 5 intervals, i.e. ticks at `[0, 2, 4, 5, 6, 10]`. Similar for 2.5 you would get ticks at `[0, 0.5, 1, 1.5, 2, 2.5]`. This does not translate well to other numbers of  divisions.

If the user sets an explicit number, he should exactly get that. Additionally, we need some `'auto'` value which triggers the current 4/5 mechanism.

Reply:
That's basically my suggestion above -- I guess "auto" is a nicer spelling than "None", though.

Reply:
Case 1:

- `Autominorticker(n=none)` and `rcParams[xtick.minor.ndivs] = none` then the default n = 4 or 5 case.

- `Autominorticker(n=none)` and `rcParams[xtick.minor.ndivs] = some value` then set n = some value.

- `Autominorticker(n=somevalue)` and `rcParams[xtick.minor.ndivs] = some value`  or `Autominorticker(n=somevalue)` and `rcParams[xtick.minor.ndivs] = none` then set n = value in argument of Autominorticker.

- Allowed value of rcParams = {none, int}

Case 2:

- `Autominorticker(n=none, auto=True)` then use `n = rcParams[xtick.minor.ndivs] `.

- `Autominorticker(n=none, auto=False)` then use 'n = 4 or 5'.

- `Autominorticker(n=somevalue, auto=False)` then use 'n = somevalue'.

- `Autominorticker(n=somevalue, auto=True)` then also 'n = somevalue'.

- Allowed value of rcParams = {int}

which case will be better for the feature @timhoffm @anntzer .

Reply:
Thanks for working out the possibilities.
I would strongly prefer 1) (no need to add a new parameter), let's see what @timhoffm says.

Reply:
This is one functionality, and this should be configured by a single parameter. Having multiple parameters with overlapping meaning gets us into trouble regularly.

I would slightly prefer an additional `'auto'` value:

- Allowed value of rcParams: {'auto', int}
- Allowed value of Autominorticker(n): {'auto', int, None}

with:
- None: use rcParams
- int: use explicitly that many ticks
- 'auto': automatically choose a good number of ticks (the current 4/5 mechanism)

This has the advantage of a clear distinction between "fall back to rc" and "find a reasonable value", which are semantically mixed in the proposal 1) above.

Reply:
I need a little help. How can i know on which axis the AutoMinorLocator() is called 
Thank you

Reply:
You can look up `self.axis.__name__`.